src/app/server/Dnssd.h

```diff
@@ -168,11 +168,14 @@ class DLL_EXPORT DnssdServer
     int16_t mDiscoveryTimeoutSecs                 = CHIP_DEVICE_CONFIG_DISCOVERY_TIMEOUT_SECS;
     System::Clock::Timestamp mDiscoveryExpiration = kTimeoutCleared;
 
+    Optional<int32_t> mExtendedDiscoveryTimeoutSecs = NullOptional;
```

Comment:
```suggestion
#if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY    Optional<int32_t> mExtendedDiscoveryTimeoutSecs = NullOptional;
#endif // CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
```

Reply:
This breaks binary compatibility. We should avoid any structural changes based on these... Too bad we already have so many

Reply:
I'd like to avoid the binary compat complications here, so will leave this unconditionally present.